tests/frontend/test_engine.py

```diff
@@ -146,13 +157,211 @@ def inspect():
             ops.Rgate(r) | q[1]
 
         eng.run(p2)
-        expected2 = expected1 + [
-            "Run 1:",
-            "Rgate({}) | (q[1])".format(r),
-        ]
+        expected2 = expected1 + ["Run 1:", "Rgate({}) | (q[1])".format(r)]
         assert inspect() == expected2
 
         # reapply history
         eng.reset()
         eng.run([p1, p2])
         assert inspect() == expected2
+
+
+class TestStarshipEngine:
+    """
+    Tests various methods on the remote engine StarshipEngine.
+    """
+
+    def test_init(self, monkeypatch):
+        """
+        Tests that a StarshipEngine instance is correctly initialized when additional APIClient
+        parameters are passed.
+        """
+        mock_api_client = MagicMock()
+        monkeypatch.setattr("strawberryfields.engine.APIClient", mock_api_client)
+        engine = StarshipEngine()
+        assert engine.client == mock_api_client()
+        assert engine.jobs == []
+
+    def test_reset(self, starship_engine):
+        """
+        Tests that StarshipEngine.jobs is correctly cleared when callling StarshipEngine.reset.
+        """
+        starship_engine.jobs.append(MagicMock())
+        assert len(starship_engine.jobs) == 1
+        starship_engine.reset()
+        assert len(starship_engine.jobs) == 0
+
+    def test__get_blackbird(self, starship_engine, monkeypatch):
+        """
+        Tests that StarshipEngine._get_blackbird returns the correct string given name,
+        shots, and program parameters.
+        """
+        methods = MagicMock()
+        inputs = MagicMock()
+
+        monkeypatch.setattr("strawberryfields.engine.to_blackbird", methods.to_blackbird)
+
+        output = starship_engine._get_blackbird(inputs.name, inputs.shots, inputs.program)
+
+        methods.to_blackbird.assert_called_once_with(inputs.program, version="1.0")
+        assert output._name == inputs.name
+        assert len(output._target.__setitem__.call_args_list) == 2
+        assert output._target.__setitem__.call_args_list[0] == call(
+            "name", starship_engine.backend_name
+        )
+        assert output._target.__setitem__.call_args_list[1] == call(
+            "options", {"shots": inputs.shots}
+        )
+
+    def test_queue_job(self, starship_engine, monkeypatch):
+        mock_job = MagicMock()
+        monkeypatch.setattr("strawberryfields.engine.Job", mock_job)
+        mock_job_content = MagicMock()
+
+        result = starship_engine._queue_job(mock_job_content)
+        mock_job.assert_called_once_with(client=starship_engine.client)
+        result.manager.create.assert_called_once_with(circuit=mock_job_content)
+        assert starship_engine.jobs == [result]
+
+    def test__run_program(self, starship_engine, monkeypatch):
+        """
+        Tests StarshipEngine._run_program. Asserts that a program is converted to blackbird code,
+        compiled into a job content string and that the job is queued. Also asserts that a
```

Comment:
the mocking makes it a bit hard to parse, but I couldn't see where it is asserted that the program is compiled into a job content string and queued.